src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Http/HttpClientFactory.cs

```diff
@@ -28,28 +28,38 @@
 using Microsoft.IdentityModel.Clients.ActiveDirectory;
 using System.Net.Http;
 using System.Net.Http.Headers;
+#if iOS
+using Foundation;
+using UIKit;
+#endif
 
 namespace Microsoft.Identity.Core.Http
-{ 
+{
     internal class HttpClientFactory : IHttpClientFactory
     {
         private readonly HttpClient _httpClient;
 
         // The HttpClient is a singleton per ClientApplication so that we don't have a process wide singleton.
-        public const long MaxResponseContentBufferSizeInBytes = 1024*1024;
+        public const long MaxResponseContentBufferSizeInBytes = 1024 * 1024;
 
         public HttpClientFactory()
         {
-            if (_httpClient == null)
+#if iOS
+            if (UIDevice.CurrentDevice.CheckSystemVersion(7, 0))
             {
-                _httpClient = new HttpClient(new HttpClientHandler() { UseDefaultCredentials = true })
+                _httpClient = new HttpClient(new NSUrlSessionHandler())
```

Comment:
Please add a comment with our learnings from Xamarin about this. You might also want to update the wiki.

Reply:
done. to keep the comment less, i just added in the aka.ms link there.